tasks/a11y/definition.py

```diff
@@ -0,0 +1,63 @@
+import os
+import shutil
+import subprocess
+
+from scraper.spider import process, A11ySpider
+
+from lib.task import BaseBuildTask
+
+
+class BuildTask(BaseBuildTask):
+    def __init__(self):
+        super().__init__(
+            extra_args=[
+                ['-t', '--target'],
+                ['-b', '--buildid'],
+                ['-o', '--owner'],
+                ['-r', '--repository']]
+        )
+
+    def handler(self):
+        """scan"""
+        target = self.args['target']
+        buildid = self.args['buildid']
+        owner = self.args['owner']
+        repository = self.args['repository']
+
+        results_dir = '/build-task/results'
+        reports_dir = '/build-task/reports'
+        templates_dir = '/build-task/reporter/templates'
+        os.makedirs(results_dir, exist_ok=True)
+
+        # crawl
+        data = []  # accumulates the urls during .start()
+        process.crawl(A11ySpider, target=target, data=data)
+        process.start()
+
+        # scan
+        for url in data:
+            subprocess.run([
+                f'axe {url}' +
+                ' --chrome-options="no-sandbox,disable-setuid-sandbox,disable-dev-shm-usage"' +
+                ' --tags wcag2a,wcag2aa,wcag21a,wcag21aa,wcag22aa' +
+                f' --dir {results_dir}'
+            ], timeout=900, shell=True)
+
+        # report
+        subprocess.run([
+            'node',
+            'build-task/reporter/generate-report.js',
```

Comment:
I'm not positive but I think if the subprocess errors it will still error this whole script and trigger the error status correctly (and print stderr to the cf logs output). That's at least what I've seen so far, including errors in this script

Reply:
I'm wrong! The error that was triggered was following the subprocess, I'll add a new common method for calling subprocess that implements bubbling up the error